### ISSUE TITLE
Deployment on K8S 1.16.x

### ISSUE THREAD

**Issue by ilijamt** [ansible/awx#5286]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
Running the installer to deploy to K8S doesn't work for creating the apps as apps/v1beta1 has been deprecated in 1.16

##### ENVIRONMENT
* AWX version: 9.0.1
* AWX install method: `ansible-playbook -i inventory install.yml`
* Ansible version:  latest

##### STEPS TO REPRODUCE

Inventory
```
localhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"

[all:vars]
dockerhub_base=ansible
kubernetes_context=ctx
kubernetes_namespace=awx
tiller_namespace=kube-system
pg_persistence_storageClass=local-path
kubernetes_ingress_hostname=awx.example.com
awx_task_hostname=awx
awx_web_hostname=awxweb
postgres_data_dir="~/.awx/pgdocker"
host_port=80
host_port_ssl=443
docker_compose_dir="~/.awx/awxcompose"
pg_username=awx
pg_password=24g52lk45
pg_database=awx
pg_port=5432
rabbitmq_password=34f135gi45ogiwj45g
rabbitmq_erlang_cookie=asfdg3og24o5g
admin_user=admin
admin_password=sdg235g245g245g
create_preload_data=False
secret_key=245g2k4j5gn245g
```
Run the installer on a blank K8S v1.16.2 built with Rancher 0.3.2

```
---
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  name: awx
  namespace: awx
spec:
  serviceName: awx
  replicas: 1
  template:
    metadata:
      labels:
        name: awx-web-deploy
        service: django
        app: awx
    spec:
      serviceAccountName: awx
      terminationGracePeriodSeconds: 10
      containers:
        - name: awx-web
          image: "ansible/awx_web:9.0.1"
          imagePullPolicy: Always
          ports:
            - containerPort: 8052
          volumeMounts:
            - name: awx-application-config
              mountPath: "/etc/tower/settings.py"
              subPath: settings.py
              readOnly: true

            - name: awx-nginx-config
              mountPath: /etc/nginx/nginx.conf
              subPath: nginx.conf
              readOnly: true

            - name: "awx-application-credentials"
              mountPath: "/etc/tower/conf.d/"
              readOnly: true

            - name: awx-secret-key
              mountPath: "/etc/tower/SECRET_KEY"
              subPath: SECRET_KEY
              readOnly: true

          resources:
            requests:
              memory: "1Gi"
              cpu: "500m"
        - name: awx-celery
          securityContext:
            privileged: true
          image: "ansible/awx_task:9.0.1"
          command:
            - /usr/bin/launch_awx_task.sh
          imagePullPolicy: Always
          volumeMounts:
            - name: awx-application-config
              mountPath: "/etc/tower/settings.py"
              subPath: settings.py
              readOnly: true

            - name: "awx-application-credentials"
              mountPath: "/etc/tower/conf.d/"
              readOnly: true

            - name: awx-secret-key
              mountPath: "/etc/tower/SECRET_KEY"
              subPath: SECRET_KEY
              readOnly: true
          env:
            - name: AWX_SKIP_MIGRATIONS
              value: "1"
          resources:
            requests:
              memory: "2Gi"
              cpu: "1500m"
        - name: awx-rabbit
          image: "ansible/awx_rabbitmq:3.7.15"
          imagePullPolicy: Always
          ports:
            - name: http
              protocol: TCP
              containerPort: 15672
            - name: amqp
              protocol: TCP
              containerPort: 5672
          livenessProbe:
            exec:
              command:
              - /usr/local/bin/healthchecks/rabbit_health_node.py
            initialDelaySeconds: 30
            timeoutSeconds: 10
          readinessProbe:
            exec:
              command:
              - /usr/local/bin/healthchecks/rabbit_health_node.py
            initialDelaySeconds: 10
            timeoutSeconds: 10
          env:
            - name: MY_POD_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
            - name: RABBITMQ_USE_LONGNAME
              value: "true"
            - name: RABBITMQ_NODENAME
              value: "rabbit@$(MY_POD_IP)"
            - name: RABBITMQ_ERLANG_COOKIE
              valueFrom:
                secretKeyRef:
                  name: "awx-secrets"
                  key: rabbitmq_erlang_cookie
            - name: K8S_SERVICE_NAME
              value: "rabbitmq"
            - name: RABBITMQ_USER
              value: awx
            - name: RABBITMQ_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: "awx-secrets"
                  key: rabbitmq_password
          volumeMounts:
            - name: rabbitmq-config
              mountPath: /etc/rabbitmq
            - name: rabbitmq-healthchecks
              mountPath: /usr/local/bin/healthchecks
          resources:
            requests:
              memory: "2Gi"
              cpu: "500m"
        - name: awx-memcached
          image: "memcached:latest"
          imagePullPolicy: Always
          resources:
            requests:
              memory: "1Gi"
              cpu: "500m"
      volumes:
```

The problem happens in this part of the deployment and only for this stateful set with the error of 
```
error: unable to recognize "": no matches for kind "StatefulSet" in version "apps/v1beta1"
```
Modification of the app from apps/v1beta1 to apps/v1 requires changes to the stateful set.

**Comment by ilijamt** [comment]
After applying this diff the deployment work successfully.

```
diff --git a/installer/roles/kubernetes/templates/deployment.yml.j2 b/installer/roles/kubernetes/templates/deployment.yml.j2
index 7b9efc721..d9f194223 100644
--- a/installer/roles/kubernetes/templates/deployment.yml.j2
+++ b/installer/roles/kubernetes/templates/deployment.yml.j2
@@ -131,7 +131,7 @@ userNames:
 {% endif %}
 
 ---
-apiVersion: apps/v1beta1
+apiVersion: apps/v1
 kind: StatefulSet
 metadata:
   name: {{ kubernetes_deployment_name }}
@@ -139,6 +139,9 @@ metadata:
 spec:
   serviceName: {{ kubernetes_deployment_name }}
   replicas: 1
+  selector:
+    matchLabels: 
+      app: {{ kubernetes_deployment_name }}
   template:
     metadata:
       labels:
```

Would there be any issues with this modification?

**Comment by shanemcd** [comment]
Hello. Thanks for digging into this. We still want to support older versions, so we probably need to query the k8s api prior to running `apply` and add a conditional in the deployment template based on the version we discover.

**Comment by geauxvirtual** [comment]
`apps/v1` has existed since Kubernetes 1.9. https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

**Comment by shanemcd** [comment]
OpenShift did not include Kubernetes 1.9 until version 3.9. We currently have customers on older versions and are not going to drop support for this, since adding a conditional here is trivial.

**Comment by ilijamt** [comment]
Will add a conditional that will change the deployment based on a flag, is that an acceptable solution @shanemcd ?

**Comment by ilijamt** [comment]
This can be closed, PR #5358 merged
